### PR TITLE
add inline bot commands

### PR DESCRIPTION
**Pull request by SebbyLaw**

### CODE REVIEW

File: discord/ext/alternatives/inline_bot_commands.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            for elem, value in base.__dict__.items():
+                if elem in cmds:
+                    del cmds[elem]
+                if isinstance(value, commands.Command):
+                    cmds[elem] = value
+
+        new_cls.__inline_commands__ = list(cmds.values())
+        return new_cls
+
+    def qualified_name(cls):
```
Comment (Gobot1234) [inline discord/ext/alternatives/inline_bot_commands.py:40]: Shouldn't this be a property?

Comment (SebbyLaw) [reply discord/ext/alternatives/inline_bot_commands.py:41]: you're right. fixed.

File: discord/ext/alternatives/inline_bot_commands.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+
+class InlineMeta(type):
+    def __new__(cls, *args, **kwargs):
+        new_cls = super().__new__(cls, *args)
+        cmds = {}
+        for base in reversed(new_cls.__mro__):
+            for elem, value in base.__dict__.items():
+                if elem in cmds:
+                    del cmds[elem]
```
Comment (Gobot1234) [inline discord/ext/alternatives/inline_bot_commands.py:33]: Isn't there no point deleting the old reference as you replace it below?

Comment (SebbyLaw) [reply discord/ext/alternatives/inline_bot_commands.py:33]: No. Consider the following:

```py
class Foo(commands.Bot):
    @commands.command()
    async def foo(self, ctx):
        ...

class Bar(Foo):
    async def foo(self, ctx):
        ...
```

Bar.foo overrides the old callback, but isn't a command object. The overriden callback is still registered as a command, so we must delete it.

Comment (Gobot1234) [reply discord/ext/alternatives/inline_bot_commands.py:33]: Oh right, that makes sense. 👍

**Pull request merged**